components/shopify-widget/index.tsx

```diff
@@ -3,49 +3,48 @@ import { Window } from "../ui/window";
 import Image from "next/image";
 import PsyButton from "../ui/psy-button";
 import getPOAPStatus from "@/utils/getPOAPStatus";
-import type { Address } from "viem";
 import { useAccount } from "wagmi";
 import ShopifyImageModal from "./shopify-image-modal";
 import { useEffect, useState } from "react";
 import useRedirectToShopify from "@/hooks/useRedirectToShopify";
-import { determineDiscountCodeUsage } from "@/utils/determineDiscountCodeUsage";
-
-const handleEligibilityLogic = async (address: Address | undefined) => {
-  const userHasNotUsedDiscount = await determineDiscountCodeUsage(address);
-  const userPOAPStatus = await getPOAPStatus(address);
-
-  return { userPOAPStatus, userHasNotUsedDiscount };
-};
+import useGetUserOrders from "@/hooks/useGetUserOrders";
 
 const ShopifyWidget = () => {
   const [isOpen, setIsOpen] = useState(false);
   const [userIsEligibleToClaim, setUserIsEligibleToClaim] = useState(false);
-
-  const { redirectToShopify } = useRedirectToShopify();
+  const [addressSnippet, setAddressSnippet] = useState("");
 
   const { address } = useAccount();
 
+  const { data, error } = useGetUserOrders(addressSnippet);
+
+  const { redirectToShopify } = useRedirectToShopify();
+
   const handleModal = () => {
     setIsOpen((prev) => !prev);
   };
 
   useEffect(() => {
+    if (!address) return;
+    const snippet = address.slice(2, 8);
```

Comment:
@EsterB12 is `snippet` the discount code now?

Reply:
the snippet is the wallet address fragment from the discount code since the discount code format is `PSYDAO-${ethAddress.slice(2, 8)}-${Date.now()}`;